app/Filament/Concerns/SemesterSelectForOperator.php

```diff
@@ -0,0 +1,111 @@
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace App\Filament\Concerns;
+
+use AdvisingApp\StudentDataModel\Models\Enrollment;
+use Filament\Forms\Components\Select;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\DB;
+
+trait SemesterSelectForOperator
+{
+    public static function semesterSelect(): Select
+    {
+        return Select::make('semesters')
+            ->label('Semester')
+            ->options(static::getSemesterOptions())
+            ->searchable()
+            ->multiple();
+    }
+
+    /**
+     * @return array<string, string>
+     */
+    public static function getSemesterOptions(): array
+    {
+        return Enrollment::query()
+            ->whereNotNull('semester_name')
+            ->distinct()
+            ->orderBy('semester_name')
+            ->pluck('semester_name', 'semester_name')
+            ->all();
+    }
+
+    /**
+     * @param  Builder<Model>  $query
+     *
+     * @return Builder<Model>
+     */
+    public function applyToBaseQuery(Builder $query): Builder
+    {
+        $relationshipName = $this->constraint->getRelationshipName();
+        $count = $this->settings['count'] ?? 1;
+        $semesters = $this->settings['semesters'] ?? null;
+
+        if (blank($semesters)) {
+            return parent::applyToBaseQuery($query);
+        }
+
+        $semesters = Arr::wrap($semesters);
+
+        $semesters = array_values(array_filter($semesters, fn ($semester) => filled($semester)));
+
+        $lowerSemesters = array_map('mb_strtolower', $semesters);
+
+        return $query->whereHas($relationshipName, function (Builder $query) use ($lowerSemesters) {
+            if (! empty($lowerSemesters)) {
+                $query->whereIn(DB::raw('LOWER(semester_name)'), $lowerSemesters);
+            }
```

Comment:
The semesters should never be empty at this point, so this condition is not required